cloudfront/invalidation/main.tf

```diff
@@ -0,0 +1,40 @@
+resource "null_resource" "invalidate" {
+  count = var.create ? 1 : 0
+
+  triggers = var.triggers
+
+  provisioner "local-exec" {
+    command = <<-EOT
+      set -e
+
+      export AWS_REGION=us-east-1
+
+      cat <<'EOF'
+      Invalidating ${var.distribution_id} paths:
+      ${join("\n", var.paths)}
+      EOF
+
+      invalidation_id=$(
+        aws cloudfront create-invalidation \
+          --distribution-id '${var.distribution_id}' \
+          --paths ${join(" ", [for path in var.paths : "'${path}'"])} \
+          --query 'Invalidation.Id' \
+          --output text
+      )
+
+      invalidation_status=InProgress
+
+      while [ "$invalidation_status" != "Completed" ]; do
```

Comment:
https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloudfront/wait/invalidation-completed.html

Reply:
Switched to `aws cloudfront wait` in 6fd571a